beacon_node/network/src/beacon_processor/worker/rpc_methods.rs

```diff
@@ -375,27 +375,22 @@ impl<T: BeaconChainTypes> Worker<T> {
         send_on_drop: SendOnDrop,
         peer_id: PeerId,
         request_id: PeerRequestId,
-        req: BlocksByRangeRequest,
+        mut req: BlocksByRangeRequest,
     ) {
         debug!(self.log, "Received BlocksByRange Request";
             "peer_id" => %peer_id,
-            "count" => req.count,
-            "start_slot" => req.start_slot,
+            "count" => req.count(),
+            "start_slot" => req.start_slot(),
         );
 
         // Should not send more than max request blocks
-        if req.count > MAX_REQUEST_BLOCKS {
-            return self.send_error_response(
-                peer_id,
-                RPCResponseErrorCode::InvalidRequest,
-                "Request exceeded `MAX_REQUEST_BLOBS_SIDECARS`".into(),
-                request_id,
-            );
+        if *req.count() > MAX_REQUEST_BLOCKS {
+            *req.count_mut() = MAX_REQUEST_BLOCKS;
```

Comment:
This behaviour changed on Deneb. Need to confirm what's the desired behaviour.
The error isn't accurate though:
> Request exceeded `MAX_REQUEST_BLOBS_SIDECARS`

Reply:
I agree with sean. We could maybe consider switching the max value given the fork context later on.

Reply:
Sounds good - I'll put in `MAX_REQUEST_BLOCKS_DENEB`. 

I was actually asking about whether we call `send_error_response` if `req.count` exceeds the limit 😅. I just realised it was added in #3912 for 4844, so I'll add it back in.

Reply:
Updated in https://github.com/sigp/lighthouse/pull/4432/commits/68140fa0363ba62fdb9b4b82a3a09872fe93b1a5. Thanks!

Reply:
Oh right! Yes I think in working on 4844 I realized it was spec'd out that we should error there instead of just truncating the request, I guess I never opened a PR for it to unstable but we could change it there, although it doesn't seem to be an issue for us atm